src/utils/api.js

```diff
@@ -0,0 +1,20 @@
+import axios from "axios";
+
+const instance = axios.create({
+    baseURL:"https://be-news-api-server.onrender.com/api"
+})
+
+function getAllArticles (){
+  return instance.get("/articles")
+}
+
+function getSingleArticle (article_id){
+    return instance.get(`/articles/${article_id}`)
+}
+
+function getAllCommentsByArticleId (article_id){
+    return instance.get(`/articles/${article_id}/comments`)
+}
+export default getAllArticles
+
+export {getSingleArticle, getAllCommentsByArticleId}
```

Comment:
Great to putting all the api calls in the same file. Your way of doing it is working fine but it would be better to prepare the wanted data here rather than do it in the .then() in useEffect.